File: WebClient SDK/src/default/workflow/krPermissions/krTokenToTaskHistoryViewUIExtension.ts
import { CardTypeExtensionTypes, executeExtensions, TypeExtensionContext } from 'tessa/cards';
import { tryGetFromSettings } from 'tessa/ui';
import { CardUIExtension, ICardUIExtensionContext } from 'tessa/ui/cards';
import { ViewControlViewModel } from 'tessa/ui/cards/controls';
import { KrToken } from 'tessa/workflow';

export class KrTokenToTaskHistoryViewUIExtension extends CardUIExtension {
  public async initializing(context: ICardUIExtensionContext) {
    const result = await executeExtensions(
      CardTypeExtensionTypes.makeViewTaskHistory,
      context.card,
      context.model.generalMetadata,
      this.executeInitializingAction,
      context
    );

    context.validationResult.add(result);
  }

  executeInitializingAction = async (typeContext: TypeExtensionContext) => {
    const context = typeContext.externalContext as ICardUIExtensionContext;
    const settings = typeContext.settings;
    const viewControlAlias = tryGetFromSettings<string>(settings, 'ViewControlAlias');

    context.model.controlInitializers.push(control => {
      if (control instanceof ViewControlViewModel) {
        if (control.name === viewControlAlias) {
          const token = KrToken.tryGet(context.card.info);
          const tokenParameterAlias = tryGetFromSettings<string>(settings, 'TokenParamterAlias');
          if (token && tokenParameterAlias) {
            // ...
          }
          // KrToken token = KrToken.TryGet(context.Card.Info);
          // var tokenParameterAlias = settings.TryGet<string>(CardTypeExtensionSettings.TokenParameterAlias);
          // if (token != null && !string.IsNullOrWhiteSpace(tokenParameterAlias))
          // {
          //     string tokenString = token.GetStorage().ToSerializable().ToBase64String();
          //     IViewParameterMetadata tokenMetadata = viewControl.ViewMetadata.Parameters.FindByName(tokenParameterAlias);
          //     if (tokenMetadata != null)
          //     {
          //         RequestParameter tokenParameter = new RequestParameterBuilder()
          //             .WithMetadata(tokenMetadata)
          //             .AddCriteria(new EqualsCriteriaOperator(), tokenString, tokenString)
          //             .AsRequestParameter();
          //         viewControl.Parameters.Add(tokenParameter);
          //     }
          // }
        }
      }
    });
  };
}

File: Source/Extensions/Tessa.Extensions.Default.Client/Workflow/KrPermissions/KrTokenToTaskHistoryViewUIExtension.cs
using System.Threading.Tasks;
using Tessa.Cards;
using Tessa.Extensions.Default.Shared.Workflow.KrProcess;
using Tessa.Platform;
using Tessa.Platform.Storage;
using Tessa.UI;
using Tessa.UI.Cards;
using Tessa.UI.Cards.Controls;
using Tessa.Views;
using Tessa.Views.Metadata.Criteria;

namespace Tessa.Extensions.Default.Client.Workflow.KrPermissions
{
    /// <summary>
    /// Устанавливает в параметре <see cref="CardTypeExtensionSettings.TokenParameterAlias"/> представления, указанного по ключу <see cref="CardTypeExtensionSettings.ViewControlAlias"/>, информацию о токене безопасности <see cref="KrToken"/>, выданного для карточки.
    /// </summary>
    public sealed class KrTokenToTaskHistoryViewUIExtension :
        CardUIExtension
    {
        #region Constructors

        public KrTokenToTaskHistoryViewUIExtension() { }

        #endregion

        #region Base Overrides

        /// <inheritdoc/>
        public override async Task Initializing(ICardUIExtensionContext context)
        {
            var result = await CardHelper
                .ExecuteTypeExtensionsAsync(
                    CardTypeExtensionTypes.MakeViewTaskHistory,
                    context.Card,
                    context.Model.CardMetadata,
                    ExecuteInitializingAsync,
                    context,
                    cancellationToken: context.CancellationToken);

            context.ValidationResult.Add(result);
        }

        #endregion

        #region Type extension methods

        private static Task ExecuteInitializingAsync(ITypeExtensionContext typeContext)
        {
            var context = (ICardUIExtensionContext) typeContext.ExternalContext;
            var settings = typeContext.Settings;
            var viewControlAlias = settings.TryGet<string>(CardTypeExtensionSettings.ViewControlAlias);

            context.Model.ControlInitializers.Add((control, m, r, ct) =>
            {
                if (control is CardViewControlViewModel viewControl)
                {
                    if (viewControl.Name == viewControlAlias)
                    {
                        var token = KrToken.TryGet(context.Card.Info);
                        var tokenParameterAlias = settings.TryGet<string>(CardTypeExtensionSettings.TokenParameterAlias);
                        if (token is not null && !string.IsNullOrWhiteSpace(tokenParameterAlias))
                        {
                            var tokenString = token.ToTypedJson();

                            var tokenMetadata = viewControl.ViewMetadata.Parameters.FindByName(tokenParameterAlias);
                            if (tokenMetadata is not null)
                            {
                                var tokenParameter = new RequestParameterBuilder()
                                    .WithMetadata(tokenMetadata)
                                    .AddCriteria(new EqualsCriteriaOperator(), tokenString, tokenString)
                                    .AsRequestParameter();
                                viewControl.Parameters.Add(tokenParameter);
                            }
                        }
                    }
                }

                return new ValueTask();
            });

            return Task.CompletedTask;
        }

        #endregion
    }
}
